src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -905,11 +905,7 @@ private void EmitCatchBlock(BoundCatchBlock catchBlock)
 
                 _builder.OpenLocalScope(ScopeType.Catch, exceptionType);
 
-                if (catchBlock.IsSynthesizedAsyncCatchAll)
-                {
-                    Debug.Assert(_asyncCatchHandlerOffset < 0); // only one expected
-                    _asyncCatchHandlerOffset = _builder.AllocateILMarker();
-                }
+                RecordAsyncCatchHandlerOffset(catchBlock);
```

Comment:
>RecordAsyncCatchHandlerOffset [](start = 16, length = 29)

:memo: previously catch block from async methods would always fall into the first branch, because they didn't use exception filters.
Now, both branches of this logic (with and without exception filter) potentially need to record that IL location for later use ("// The exception handler IL offset is used by the debugger to treat exceptions caught by the marked catch block as "user unhandled".")